### ISSUE TITLE
Issue with projected band plotting

### ISSUE THREAD

**Issue by khaledb9** [SMTG-Bham/easyunfold#30]
Hello 

I'm trying to plot the band structure for my system. The overall band structure works fine but any projections yield the error: 
P_Km = P_Km * band_weight_sets[ii][jj][ispin, :P_Km.shape[0]]
                          ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
IndexError: index 1 is out of bounds for axis 0 with size 1"


Kindly help.

**Comment by khaledb9** [comment]
More details

unfold.py", line 712, in spectral_function_from_weight_sets
    P_Km = P_Km * band_weight_sets[ii][jj][ispin, :P_Km.shape[0]]
                  ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
IndexError: index 1 is out of bounds for axis 0 with size 1

**Comment by zhubonan** [comment]
Hi @khaledb9 thanks for reporting it - please provide more details. Is it a single calculation with projection? Please at least post the full trace back and the `json` and PROCAR if possible. 
It looks like either the `json` file or the PROCAR have incorrect information.

**Comment by zhubonan** [comment]
@khaledb9 Can you please check if the updated code works for you now? Thanks

**Comment by khaledb9** [comment]
Hi
Sorry for the delayed response. The issue is solved and it was related to the handling of spin. Thanks !

**Comment by khaledb9** [comment]
Hello 

The issue has been solved. However, only when using the --combined option to plot projections yields the error below: 
line 336, in _plot_spectral_function_rgba
    fig = axes[0].figure
AttributeError: 'numpy.ndarray' object has no attribute 'figure'

**Comment by kavanase** [comment]
Hi @khaledb9, sorry for the delay in looking at this, our lead developer has recently started a new position so has been quite busy!

I've looked and found the issue that was causing this, and has now been fixed in e4e16514b25bbbde3b5a68bd02d0d142b3bf1cc9 and so if you update the code it should all work!
You can do this with:
```
git clone https://github.com/SMTG-Bham/easyunfold
cd easyunfold
pip install -e .
```
or if you've already cloned the repo before, can just go to the repo folder and do `git pull && pip install -e .`.

This will also be included in the next release of `easyunfold` on `PyPI` which should be quite soon!
Please let us know if this doesn't fix your issue! :smiley:

**Comment by kavanase** [comment]
Hi @khaledb9! Just to mention, this fix has been included in the latest release of `easyunfold` (`v0.3.2`), and so can be installed with `pip install --upgrade easyunfold`
